### PR TITLE
Build-time ProblemProcessor simplified /with automatic classpath detection/ so it will be easier to add more mappers

### PR DESCRIPTION
**Pull request by lwitkowski**
`+31 −113` - isn't it what we all like? ;)

### CODE REVIEW

File: deployment/src/main/java/com/tietoevry/quarkus/resteasy/problem/deployment/ProblemProcessor.java [hunk 1]
```diff
@@ -25,62 +25,59 @@
...
+
+        mappers.put("com.fasterxml.jackson.core.JsonProcessingException", "jackson.JsonProcessingExceptionMapper");
+        mappers.put("javax.ws.rs.ProcessingException", "jsonb.RestEasyClassicJsonbExceptionMapper");
+        mappers.put("javax.json.bind.JsonbException", "jsonb.JsonbExceptionMapper");
+
+        return mappers.entrySet().stream()
+                .filter(mapper -> {
+                    String exceptionClass = mapper.getKey();
+                    return classAvailable(exceptionClass);
+                })
```
Comment (Wertros) [inline deployment/src/main/java/com/tietoevry/quarkus/resteasy/problem/deployment/ProblemProcessor.java:52]: Maybe it's better to extract whole logic of lambda and put it to one method called `isClassAvailable()` or `hasRegisteredClass()` or `isOnClasspath()`
For me it's always better to have a prefix for all boolean methods i.e `is` or `has` ...

File: deployment/src/main/java/com/tietoevry/quarkus/resteasy/problem/deployment/ProblemProcessor.java [hunk 2]
```diff
@@ -25,62 +25,59 @@
     private static final String FEATURE_NAME = "resteasy-problem";
     private static final String PACKAGE = "com.tietoevry.quarkus.resteasy.problem.";
```
Comment (Wertros) [inline deployment/src/main/java/com/tietoevry/quarkus/resteasy/problem/deployment/ProblemProcessor.java:26]: Maybe `INTERNAL_PACKAGE`

**Review by Wertros — APPROVED** [review]
`+31 −113` - isn't it what we all like? - TRUE  👍

File: deployment/src/main/java/com/tietoevry/quarkus/resteasy/problem/deployment/ProblemProcessor.java (hunk 2 above)
Comment (lwitkowski) [reply deployment/src/main/java/com/tietoevry/quarkus/resteasy/problem/deployment/ProblemProcessor.java:26]: Agree, `PACKAGE` is not very descriptive. 
Another option: `EXTENSION_MAIN_PACKAGE` ?

File: deployment/src/main/java/com/tietoevry/quarkus/resteasy/problem/deployment/ProblemProcessor.java (hunk 1 above)
Comment (lwitkowski) [reply deployment/src/main/java/com/tietoevry/quarkus/resteasy/problem/deployment/ProblemProcessor.java:52]: I can add `is`, no problem. Remember though, that this method accepts either exception class name, or not-very-readable `Map.Entry`, as we're filtering map, not a list. 

Thus:`filter` accepts `Predicate<Map.Entry>`, not `Predicate<String>`. I find it more readable extracting entry.getKey into properly named variable first (`exceptionClass`), then passing it to final method.

**Pull request merged**